pandas/tests/io/excel/test_writers.py

```diff
@@ -1162,6 +1162,19 @@ def test_path_local_path(self, engine, ext):
                                        path="foo.{ext}".format(ext=ext))
         tm.assert_frame_equal(result, df)
 
+    def test_merged_cell_custom_objects(self, engine, ext):
+        # see GH-27006
+        mi = MultiIndex.from_tuples([(pd.Period('2018'), pd.Period('2018Q1')),
+                                     (pd.Period('2018'), pd.Period('2018Q2'))])
+        expected = DataFrame(np.random.rand(2, len(mi)), columns=mi)
+        expected.to_excel(self.path)
+        result = pd.read_excel(self.path, header=[0, 1], index_col=0)
+        # need to convert PeriodIndexes to standard Indexes for assert equal
+        expected.columns.set_levels([[str(i) for i in mi.levels[0]],
```

Comment:
Can just simplify this as `expected.columns = expected.columns.astype(object)` - Period won't be lossless to / from Excel so makes sense

Reply:
Simplify this as stated

Reply:
This simplication doesn't seem to work. Comparison fails - the Index read back in isn't preserved as a Period.

E           MultiIndex level [0] classes are not equivalent
E           [left]:  PeriodIndex(['2018', '2018'], dtype='period[A-DEC]', freq='A-DEC')
E           [right]: Index(['2018', '2018'], dtype='object')